src/placement/pl_map_common.c

```diff
@@ -76,21 +95,38 @@ remap_alloc_one(d_list_t *remap_list, unsigned int shard_idx,
 		struct pool_target *tgt)
 {
 	struct failed_shard *f_new;
+	int rc;
 
-	D_ALLOC_PTR(f_new);
-	if (f_new == NULL)
-		return -DER_NOMEM;
+	rc = alloc_f_shard(&f_new, shard_idx, tgt);
+	if (rc == 0) {
+		f_new->fs_tgt_id = -1;
+		remap_add_one(remap_list, f_new);
+	}
+	return rc;
+}
 
-	D_INIT_LIST_HEAD(&f_new->fs_list);
-	f_new->fs_shard_idx = shard_idx;
-	f_new->fs_fseq = tgt->ta_comp.co_fseq;
-	f_new->fs_status = tgt->ta_comp.co_status;
-	f_new->fs_tgt_id = -1;
+/**
+   * Allocate a new failed shard then add it into remap list
+   *
+   * \param[in] remap_list        List for the failed shard to be added onto.
+   * \param[in] shard_idx         The shard number of the failed shard.
+   * \paramp[in] tgt              The failed target that will be added to the
+   *                              remap list.
+   */
+int
+reint_alloc_one(d_list_t *remap_list, unsigned int shard_idx,
+		struct pool_target *tgt)
```

Comment:
do we actually need a separate function, or just add a parameter to say it's "reint" or "rebuild"